tools/websocket_to_posix_proxy/src/posix_sockets.h

```diff
@@ -40,6 +42,15 @@
 #define SEND_FORMATTING_SPECIFIER "%d"
 #define CLOSE_SOCKET(x) closesocket(x)
 
+// Winsock documentation states that the "BSD style of error checking "will not work on Windows":
+// https://docs.microsoft.com/en-us/windows/win32/winsock/return-values-on-function-failure-2
+// but instead must check against SOCKET_ERROR or INVALID_SOCKET
+
+// On Windows, API calls that return a socket should test against INVALID_SOCKET:
+// https://docs.microsoft.com/en-us/windows/win32/winsock/handling-winsock-errors
+#define CREATE_SOCKET_CALL_RETURNS_AN_ERROR(x) ((x) == INVALID_SOCKET)
+#define SOCKET_API_CALL_RETURNS_AN_ERROR(x) ((x) == SOCKET_ERROR)
```

Comment:
How about the slightly shorter `IS_SOCKET_API_ERROR` and `IS_CREATE_SOCKET_ERROR`?

Reply:
Ops, this PR fell in between.. this is a good suggestion, updated.